apps/dashboard/src/hooks/use-form-autosave.ts

```diff
@@ -18,9 +18,6 @@ export function useFormAutosave<U extends Record<string, unknown>, T extends Fie
   save: (data: U) => void;
 }) {
   const formRef = useDataRef(propsForm);
-  const watchedData = useWatch<T>({
-    control: propsForm.control,
-  });
```

Comment:
the first mentioned reason for "lagging" in the editor, `useWatch` is causing re-renders when the form state is updated